yaksh/views.py

```diff
@@ -1831,3 +1832,18 @@ def download_yaml_template(request):
     
     return response
 
+@login_required
+@email_verified
+def preview_questionpaper(request, questionpaper_id):
+    user = request.user
+    if not is_moderator(user):
+        raise Http404('You are not allowed to view this page!')
+    paper = QuestionPaper.objects.get(id=questionpaper_id)
```

Comment:
You need to add one more condition to check if the user is course creator or course teacher.

Reply:
Added a check to ensure that the user is the creator of the quiz of that question paper in the change-set in the reopened PR.